### PR TITLE
SNOW-1637102: Support binary operations between timedelta and number.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Fixes SNOW-1637102

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py [hunk 1]
```diff
@@ -385,32 +429,102 @@ def compute_binary_op_between_snowpark_columns(
...
+        "ge",
+        "lt",
+        "le",
+    ):
         # These operations, when done between timedeltas, work without any
         # extra handling in `snowpark_pandas_type` or `binary_op_result_column`.
-        # They produce outputs that are not timedeltas (e.g. numbers for floordiv
-        # and truediv, and bools for the comparisons).
         pass
+    elif op == "mul" and (
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py:443]: the readability of `compute_binary_op_between_snowpark_columns` is getting worse IMO. Just a heads up, I will make some refactoring later after the recent changes to make it more structured.

File: src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py [hunk 2]
```diff
@@ -385,32 +429,102 @@ def compute_binary_op_between_snowpark_columns(
...
+        "mod",
+        "gt",
+        "ge",
+        "lt",
+        "le",
+        "ne",
+        "eq",
+    ) and (
+        (
+            isinstance(first_datatype(), TimedeltaType)
```
[the remark this answers is not in the document]
Comment (sfc-gh-mvashishtha) [reply src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py:511]: These are the valid binary operations in pandas. Invalid operations, like bitwise xor and exponentiation, shouldn't raise `NotImplementedError` here.

I've now added test cases for those in `test_invalid_ops_between_timedelta_dataframe_and_string_scalar`.

File: src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py [hunk 3]
```diff
@@ -385,32 +429,102 @@ def compute_binary_op_between_snowpark_columns(
         )
     elif _op_is_between_two_timedeltas_or_timedelta_and_null(
         first_datatype(), second_datatype()
-    ) and op in ("eq", "ne", "gt", "ge", "lt", "le", "truediv"):
+    ) and op in (
```
[the remark this answers is not in the document]
Comment (sfc-gh-mvashishtha) [reply src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py]: Done.

File: tests/integ/modin/binary/test_timedelta.py [hunk 4]
```diff
@@ -1405,3 +1960,33 @@ def test_two_timedelta_datfaframes(
...
+        numeric_dataframes_postive_no_nulls_1_2x2,
+        timedelta_dataframes_postive_no_nulls_1_2x2,
+        op_between_numeric_and_timedelta,
+    ):
+        snow_lhs, pandas_lhs = numeric_dataframes_postive_no_nulls_1_2x2
+        snow_rhs, pandas_rhs = timedelta_dataframes_postive_no_nulls_1_2x2
+        eval_snowpark_pandas_result(
+            (snow_lhs, snow_rhs),
+            (pandas_lhs, pandas_rhs),
+            lambda t: getattr(t[0], op_between_numeric_and_timedelta)(t[1]),
```
Comment (sfc-gh-nkrishna) [inline tests/integ/modin/binary/test_timedelta.py:1991]: This is pretty neat, didn't know `eval_snowpark_pandas_result` could handle this

**Review by sfc-gh-nkrishna — APPROVED** [review]
Nice work, LGTM!

**Review by sfc-gh-azhan — APPROVED** [review]

**Pull request merged**